showcase/src/main/kotlin/com/recco/showcase/main/MainScreen.kt

```diff
@@ -38,9 +43,52 @@ import com.recco.showcase.ui.theme.Typography
 import com.recco.showcase.ui.theme.WarmBrown
 
 @Composable
-fun MainScreen(
+fun MainRoute(
     logoutClick: () -> Unit,
     openReccoClick: () -> Unit,
+    navigateToCustomizeScreen: () -> Unit,
+    viewModel: MainViewModel = hiltViewModel()
+) {
+    val application = LocalContext.current.applicationContext as Application
+    val uiState by viewModel.viewState.collectAsStateWithLifecycle()
+
+    MainScreen(
+        uiState = uiState,
+        logoutClick = logoutClick,
+        openReccoClick = openReccoClick,
+        selectionClickPalette = { palette ->
+            viewModel.setReccoPalette(palette)
+
+            ShowcaseApp.initSDK(
+                application,
+                ReccoStyle(
+                    font = viewModel.selectedReccoFont,
+                    palette = viewModel.selectedReccoPalette
```

Comment:
maybe we can use directly the `palette` returned by the lambda.
The same for the font below.
For me this is easier to understand that trusting in that the VM will do the operation synchronously, which could be error-prone when, for example, we decide to do asynchronous operations for storing the selected value

Reply:
I prefer to keep as it is because by doing this we are already using the persisted value that will be used later on when the app memory gets destroyed. So, if there is any issue, we can see it right away 👍

Reply:
I didn't mean to get rid of the `viewModel.setReccoPalette(palette)`, only change this line,
```kotlin
palette = palette
```